torchvision/csrc/io/image/cuda/encode_decode_jpeg_cuda.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <torch/types.h>
+#include "../image_read_mode.h"
+
+#if NVJPEG_FOUND
+#include <nvjpeg.h>
+
+extern nvjpegHandle_t nvjpeg_handle;
+extern std::once_flag nvjpeg_handle_creation_flag;
+#endif
+
+namespace vision {
+namespace image {
+
+C10_EXPORT torch::Tensor decode_jpeg_cuda(
+    const torch::Tensor& data,
+    ImageReadMode mode,
+    torch::Device device);
+
+C10_EXPORT std::vector<torch::Tensor> encode_jpeg_cuda(
```

Comment:
Just chiming in, these are good questions

We need to return a vector instead of a stacked tensor because the encoded jpegs aren't necessarily of the same length. We wouldn't be able to fit those in a stacked tensor since by definition all the tensors in a stack must have the same shape.

As for why we're using a vector for the *input*: this is for similar reason, i.e. the input images may not all be of the same shape. I do think we should allow the Python API (`image.io.decode_jpeg()`) to support a stacked tensor of shape `NCHW`, and possibly convert that internally to a vec of vectors of shape `CHW` so it can be passed to this C++ function, but that can be left out for another PR and I didn't want to bother Dominik with that right now.

Reply:
Nicolas beat me to it, but yes, the reason for the vector is due to heterogeneous input and output tensor shapes.